### PR TITLE
Robot

### PR DESCRIPTION
**Pull request by Licini**
Adding support for robot.
- Automatically create links in a hierarchical fashion
- Expose function for rotating specific joints.
image

### CODE REVIEW

File: src/compas_view2/objects/robotobject.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from compas.geometry import Circle
+from compas.geometry import Transformation
+from compas.geometry import Translation
+from compas.geometry import Rotation
+from compas.robots import Geometry
+from math import radians
+
+
+class RobotObject(CollectionObject):
+    """Object for displaying COMPAS RobotModel."""
```
Comment (brgcode) [inline src/compas_view2/objects/robotobject.py:17]: params, attributes, example

File: src/compas_view2/objects/robotobject.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def __init__(self, robot, **kwargs):
+        super().__init__(Collection(), name=robot.name, **kwargs)
+        self.robot = robot
+        self.joints = {}
+        self.link_objs = {}
+        self.create(robot.root, self)
+
+    def create(self, link, parent, parent_joint=None):
```
Comment (brgcode) [inline src/compas_view2/objects/robotobject.py:26]: docstring

File: src/compas_view2/objects/robotobject.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                'joint': parent_joint,
+                'axis': parent_joint.axis.vector
+            }
+
+        self.link_objs[link.name] = obj
+
+        for joint in link.joints:
+            self.create(joint.child_link, parent=obj, parent_joint=joint)
+
+    def rotate_joint(self, joint_name, angle):
```
Comment (brgcode) [inline src/compas_view2/objects/robotobject.py:57]: docstring

File: src/compas_view2/objects/object.py [hunk 4]
```diff
@@ -168,18 +168,27 @@ def _update_matrix(self):
             for child in self.children:
                 child._update_matrix()
 
+    @property
```
Comment (brgcode) [inline src/compas_view2/objects/object.py:171]: document in class docsting

File: src/compas_view2/objects/object.py [hunk 5]
```diff
@@ -63,8 +63,9 @@ def build(data, **kwargs):
             raise TypeError("Type {} is not supported by the viewer.".format(type(data)))
         return obj
 
-    def __init__(self, data, name=None, is_selected=False, is_visible=True):
+    def __init__(self, data, app=None, name=None, is_selected=False, is_visible=True):
```
Comment (brgcode) [inline src/compas_view2/objects/object.py:66]: update class docstring?

**Comment by Licini** [comment]
Updated the docstring

**Pull request merged**